<html>

<style>
#nav-body {
	font-family: Arial, Helvetica, sans-serif;
	background-color: white;
	display: inline-block;
	width: 100%;
	vertical-align: top;
	overflow: hidden;
}
#menu {
	margin: 0;
	padding: 0;
	width: 12em;
	background-color: white;
	list-style-image: none;
	list-style-type: none;
	overflow: auto;
	position: relative;
	display: inline-block;
	clear: both;
	float: left;
	width:300px;
}
#menu li {
	font: 90%"Arial", "Bitstream Vera Sans", "Trebuchet Unicode MS", "Lucida Grande", Verdana, Helvetica, sans-serif;
	margin: 0;
	padding: 0;
	list-style: none;
}
#menu a {
	background: white;
	border-bottom: 0px solid #336699;
	color: #000;
	display: block;
	margin: 0;
	padding: 8px 12px;
	text-decoration: none;
	font-weight: normal;
}
#menu a:hover {
	background: #2580a2 url("images/hover.gif") left center no-repeat;
	color: #fff;
	padding-bottom: 8px;
}
#content-wrapper {
	width: auto;
	overflow: hidden;
}

.content {
	font-size: medium;comment in css
	color: black;
	z-index:-1;
	background-color: #FFF;
	display: inline-block;
	width: 100%;
	height: 88%;
	vertical-align: 10%;
	overflow: auto;
	overflow-y: scroll;
	/*
	height:490px;
	*/
}

.content::before1{
		content:"";
		background-size: cover;
		position: absolute;
		top: 0px;
		right: 0px;
		bottom: 0px;
		left: 0px;
		background-image: url("MINNO Logo LRG_centered.svg");
		background-repeat: no-repeat;	
		background-attachment: fixed;
		background-position: center;
		background-size: 800px;
		opacity:0.10;
		z-index: 0;
		
		/*
		background-size: 150px;
		width: 300px;
		height: 300px;
		background-position: 25% 75%;
		background-size: cover;
		*/			
	}
	
	.header img {
		float: left;
		width: 200px;
		height: 100px;
		padding-left: 2%;
		background: #fff;
	}

	.header h1 {
		position: relative;
		top: 18px;
		left: 10px;
	}
	
	.offset:before { 
		display: block; 
		content: " "; 
		height: 00px;      /* Give height of your fixed element */
		margin-top: -0px; /* Give negative margin of your fixed element */    
		visibility: hidden; 
	}
	
	strong2{
	  font-weight:bold;
	  font-size:28px;
	  line-height:22px;
	}

</style>

<body>

<div id="nav-body">
	<div class="header">
		<a href="https://lewisresearchgroup.github.io/MINNO/">
		<img src="MINNO Logo LRG_centered.svg" alt="logo" />
		<h1 align="center">
		<font face="Arial" color="black" size="6"> 
			Metabolic Interactive Nodular Network for Omics <br> User Manual 
		</font>
		</h1>
		</a>
	</div>
	

  <div id="menu">

    <ul>
      <li class='active'><a href='#intro'>Introduction</a>
      </li>
      <li><a href='#databases'>Databases</a>
      </li>
      <li><a href='#tutorial'>Tutorial</a>
		<ul>
			<li><a href='#basic'>Basic</a>
			</li>
			<li><a href='#gene_knockout'>Gene Knockout </a>
			</li>
			<li><a href='#protein_exp'>Protein Expression </a>
			</li>
			<li><a href='#crossfeeding'>Cross-feeding</a>
			</li>
		</ul>
      </li>
      <li><a href='#upload'>Upload data</a>
      </li>
      <li><a href='#download'>Download data</a>
      </li>
      <li><a href='#schema'>Datafile Schema</a>
      </li>
      <li><a href='#visual'>Visualization</a>
      </li>
      <li><a href='#features'>Features</a>
      </li>
      <li><a href='#edit_tools'>Edit tools</a>
      </li>
      <li><a href='#copyright'>Copyright</a>
      </li>
    </ul>
  </div>
  <!--end menu div-->

  <div id="content-wrapper">
    <div class="content" left="50px">
		<p><a class="offset" name="intro"></a>
					<strong2>Introduction:</strong2>
					<br><br>
					&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
					<font face="Arial" size="3">
						
						<strong>Metabolic Interactive Nodular Network for Omics (MINNO)</strong> is a <a href="https://d3js.org/" target="_blank"><span>D3 javascript library</span></a> based web application to visualize biological data onto metabolic networks. Besides visualization, this tools offer various ways to share data and results to facilitate collaboration across multiple research groups globally. Being a web-application, it does not require any installation and it can be run on both Chrome and Mozilla browsers. MINNO comes with 66 base metabolic pathways from the KEGG database, which cover all primary metabolic pathways and can be combined together to build any possible large-scale metabolic network as needed. Beside KEGG, the tool also has access to the metabolic network models from BiGG Models database. MINNO enables users to built organism specific metabolic map by superimposing large-scale metabolic network formed from KEGG database with organism’s known metabolic pathways data without need to rebuild a network from scratch for each organism. The BiGG models already comes with organism specific models, so need to superimpose any metabolic maps. However, user need to organize the network as BiGG models do not come with prelayout network maps.
						
						<br><br>
						
						The tool accepts any kind of omics data, including metabolomics, proteomics, and fluxomics data as long as they can be visualized by changing sizes of nodes and edges of the metabolic network. The modular design promotes accelerated investigation of omics data on genome-scale metabolic networks for multiple species.
						
						<br><br>
						
						<a class="offset" name="databases"></a> 
						<strong2>Databases:</strong2>
						<br><br>
						MINNO comes with two biological databases: KEGG (Kyoto Encyclopedia of Genes and Genomes) and BiGG (Biochemical, Genetic and Genomics) databases.
						
						<ol>
							<li><a href="https://www.kegg.jp/" target="_blank"><span><strong>KEGG-Database:</strong></span></a> KEGG (Kyoto Encyclopedia of Genes and Genomes) is a knowledge base for systematic analysis of gene functions, linking genomic information with higher order functional information stored in the PATHWAY database, which contains graphical representations of cellular processes, such as metabolism, membrane transport, signal transduction and cell cycle. The KEGG database harbour metabolic pathways of 860 eukaryotes and 7,934 prokaryotes, of which 410 are archaea making it an excellent database to explore cross-species interaction in diverse environmental conditions and discover unknown links and interactions between biochemicals or species. MINNO contains 66 pre-layout metabolic pathways for quick use and data visualization and about 150 in total metabolic pathways for metabolism. Remaining non-metabolic pathways cannot be currently visualized on the tool. The menu contains two categories: 1) selected Ortholog/base maps which contains 66 pre-layout metabolic pathways 2) list of pathways which contains all the metabolic pathways. User can also access all the pathways documented for specific user selected organism and superimpose on base maps to obtain organism’s specific metabolic network. 
							</li>
							
							<br>
							
							<li> <a href="http://bigg.ucsd.edu/" target="_blank"><span><strong>BiGG-Models:</strong></span></a> BiGG Models database is a knowledgebase of genome-scale metabolic network reconstructions. BiGG Models integrates more than 70 published genome-scale metabolic networks into a single database with a set of stardized identifiers called BiGG IDs. Genes in the BiGG models are mapped to NCBI genome annotations, and metabolites are linked to many external databases (KEGG, PubChem, and many more). </li>
							
						</ol>
						
				</p>
				
				<p>
						<font face="Arial" size="3"> <a class="offset" name="tutorial"></a>
						
							<strong2>Tutorial:</strong2>
							<br><br>
							This tutorial is broken into 4 sections. <br><br>
							<!--
							<a href='#basic'>Basic</a> <br>
							<a href='#gene_knockout'>Gene Knockout</a> <br>
							<a href='#protein_exp'>Protein Expression</a> <br>
							<a href='#crossfeeding'>Cross-feeding</a> <br>
							-->
							
							<a class="offset" name="basic"></a>
							<strong3>Basic:</strong3>
							<br><br>
							 
							This tutorial showcases a sample dataset containing intracellular, extracellular, and flux metabolomics data from two organisms: Staphylococcus aureus (SA) and Pseudomonas aeruginosa (PA). Throughout this tutorial, you will gain knowledge on downloading a basic network pathway from the KEGG database, incorporating supplementary reactions into the network, and uploading a dataset onto your customized network.
							 
							<a href="2022_07_15_MINNO_Supplemental_Sample_Data.csv" download>
							  Download sample dataset
							</a>
							<ul>
								<li><strong>Part 1: Downloading the ortholog network:</strong> 
									<br>
									
									<ol>
										<br>
										<li>Download the sample dataset and save it in a new folder.</li>
										<br>
										<li>Click the "Simulation ON" button, which will update to display "Simulation OFF" that will retain the (X,Y) coordinates of the network layout. To access network files, click on the drop-down menu labeled 'KEGG Database'. </li> 
										<br>
										<img src="images/Screenshot from 2022-09-28 00-56-58.png" width=750 />
										<br><br>
										<li> Enter 'Arginine biosynthesis' in the search bar and choose 'Arginine Biosynthesis' from the Selected Ortholog Maps category.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 00-57-37.png" width=750 />
										<br><br>
										<li>Click on the Select Rxns drop-down menu and click 'Select All'. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 00-57-54.png" width=750 />
										<br><br>
										<li> Now, the Arginine Biosynthesis network should be visible. You can adjust the position of the network on the screen by dragging it with your cursor. Additionally, you can zoom in or out using the scroll function on your mouse. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 00-58-21.png" width=750 />
										<br><br>
										<li> To export the network in JSON format, right-click on the screen. From the options that appear, select "Export," then choose "Network," and finally click on "JSON." Save the exported file in JSON format in the same folder as the sample dataset. This is an ortholog or base network  file. It is important to save the file with the prefix "Ortholog_ko_". MINNO tool will automatically suggest a suitable name for the file, and it is recommended to use the suggested name. If the file name already exists, you can add a convenient suffix to differentiate it. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 00-58-44.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								<li><strong>Part 2: Downloading the organism network:</strong> 
								<br>
									<ol>
										<br>
										<li> Now, click on the "KEGG Organisms" list drop-down menu. Search for "Pseudomonas aeruginosa" and select PA01 strain.  </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-00-27.png" width=750 />
										<br><br>
										<li>After selecting the specific species strain (PA01), we need to specify the metabolic pathway. Click on the "KEGG Database" drop-down menu and search for "Arginine Biosynthesis". Under the "List of Pathways" category, select "Arginine biosynthesis". </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 08-36-53.png" width=750 />
										<br><br>
										<li>A new window will pop up with an XML file. This is the KGML file from the KEGG database titled "Arginine biosynthesis" of organism "pae", which is the KEGG ID for Pseudomonas aeruginosa. Right-click on the file and select the "Save as" option to save the file in the same folder as the other files. </li>
										<br>
										<li> Now, upload the XML file that you just saved. If the "Simulation ON" option is active, the nodes of the network will self-organize based on the D3.js library. If not, the network will remain static. Right-click on the screen and select "Export > Network > JSON". This is an organism-specific KEGG file. The file name should have the prefix "KEGG_org_" where "org" represents the organism ID. MINNO tool will automatically suggest a suitable name for the file, and it is recommended to use the suggested name. If the file name already exists, you can add a convenient suffix to differentiate it. Save this file in your tutorial folder. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-03-08.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								<li><strong>Part 3: Combining the ortholog and organism-specific network/s:</strong> 
								<br>
									<ol>
										<br>
										<li> Refresh MINNO, ensuring that the simulation is off to prevent any alterations to the network layout due to node movements. Right-click and select Import > Network. Choose both the Ortholog and KEGG networks you created in the folder (in .json format only). Wait for the reaction menu to update with the newly uploaded files. Click on Select Rxns > Select all, and the network will appear on the screen. All the annotated reactions for the PA01 metabolic network in the KEGG database will be displayed as nodes connected by black lines/edges, while unannotated reactions will be represented by light grey lines/edges between nodes. Right-click to export this file in JSON format as before, and save it as "KEGG_pae_Arginine biosynthesis_combined.json" in your tutorial folder. You can choose any arbitrary name as long as it allows you to identify it as the Arginine biosynthesis network of Pseudomonas aeruginosa. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-04-01.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								<li><strong>Part 4: Adding reactions to the network:</strong>
									<ol>
										<br>
										<li> In case you need to add a reaction for the transport of extracellular metabolites such as L-Arginine, right-click on the screen and select Insert > Reaction.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-04-25.png" width=750 />
										<br><br>
										<li> To view the reaction node name, click on the Grid, Tooltip drop-down menu and unselect "Reaction node name." This will display the reaction names.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-05-52.png" width=750 />
										<br><br>
										<li> Double-click on the text beside the new reaction nodes to change their names. For the grey reaction node, change the name to "L-Arginine outflux," and for the white metabolite node, change the name to "L-Arginine." You do not need to change the Subsystem or gene names, but you can add them to your future visualizations if desired. 
										</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-07-32.png" width=750 />
										<br><br>
										<li> Click and drag to re-arrange the reaction node so that the white metabolite node is on top of the reaction node. For alignment, press the SHIFT key and drag your cursor over the nodes from L-Arginine down to N-acetyl-L-citrulline in the original network. Once those nodes are all highlighted in green, release the SHIFT key and right-click. From the menu, select Geometric operations > Alignment > Vertical. This will align the selected nodes vertically. If you make a mistake with the alignment, you can use the right-click menu to select the Undo option. 
										</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-08-16.png" width=750 />
										<br><br>
										<li>To connect your new outflux reaction to the rest of the network, hold down the CTRL key and left-click on the L-Arginine node of the original network. Then, while still holding CTRL, drag your cursor to the L-Arginine outflux grey reaction node and release the mouse button. A black line connecting the two nodes will appear. If you accidentally drag over some text while performing this step, you may need to repeat the process to ensure the connection is made correctly. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-08-38.png" width=750 />
										<br><br>
										<li>Export this new network by right-clicking on the screen and selecting Export > Network > JSON. Save the file in your tutorial folder with the name "KEGG_pae_Arginine biosynthesis_combined_rxn.json". After saving, refresh your browser and click on the Simulation ON button to preserve the network layout. Next, re-upload your new network file by selecting Import > Network from the right-click menu. Choose the JSON file you just saved and then select all the reactions under the Select Rxns menu. The network with your outflux reaction should now appear. Note that by default, only the metabolite node names are displayed, but you can un-select "Reaction node names" on the Grid, Tooltip drop-down menu to view or edit the reaction names.  </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-09-04.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								<li><strong>Part 5: Uploading data to the network:</strong> 
									<ol>
										<br>
										<li> Right-click on the screen and select Import > Datafile. Choose the tutorial data file that you saved at the beginning. To classify different metabolite nodes based on their groups, press and hold the SHIFT key, and then drag your cursor over the intracellular nodes, excluding the recently added L-Arginine node. Once the desired nodes are highlighted in green, right-click on the screen and select Select > Node from the menu.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-09-30.png" width=750 />
										<br><br>
										<li>Click on the Group drop-down menu, which is currently displaying "extracellular". Select "intracellular" from the menu. Then, left-click on the network screen to deselect those nodes. Repeat the same process for the extracellular node, which is the L-Arginine node that you added manually. Make sure to only highlight the white metabolite node, not the grey reaction node. From the Group drop-down menu, select "extracellular". </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 08-46-07.png" width=750 />
										<br><br>
										
										<li> To normalize the data for each group, click on the Methods drop-down menu. It should currently be displaying "None". Then, click on "Control Normalization" under the extracellular subheading.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 08-50-46.png" width=750 />
										<br><br>
										<li> Next, click on the Parameters drop-down menu and select "scaling". Drag the larger slider bar to your desired scaling value, in this case, 2. Repeat this process for the intracellular subheading by selecting "Control Normalization". To register the scaling factor, slide the bar away from 2, release it, and then slide it back to 2 again. Remember to adjust the parameters for each group separately. Under the Link Group section, select "None". For the scaling value, adjust it to 1 as these data are already normalized and do not require additional normalization steps. </li>
										
										<br>
										
										<li>Now, click on "Invert" on the color bar at the bottom of the menu. This will make blue color represent low values while red color represents high values. Please note that for your own visualizations, you can choose any color scheme from the color-scheme drop-down menu that suits your preferences.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-37-34.png" width=750 />
										<br><br>
										<li>To visualize the Staphylococcus aureus (SA) data, click on the menu at the top of the screen that currently displays "PA" and select "SA" instead. The nodes on the screen that have SA data should now be colored accordingly, and the outflux reaction should have a bolded arrow indicating the flux. You can follow the same instructions to access the S. aureus data on a different tab in your browser and display the data there. This way, you can compare the data from both species at each time point. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 09-22-32.png" width=750 />
										<br><br>
										<li>To change the timepoint to 6hr, select the menu that currently displays "12hr" and choose "6hr" instead. This will update the visualization to show the data for the selected timepoint. For cleaner final visualizations, you can select "Link node" from the Grid, Tooltip drop-down menu. This will remove the grey link nodes from your network, while still keeping the white metabolite nodes visible. This can enhance the clarity of your network visualization.</li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-39-19.png" width=750 />
										<br><br>
										<li>To practice some of these features, you can try adding another reaction to the network. First, refresh your browser to remove any existing data from the network. Then, follow the same steps you used to add the L-Arginine outflux reaction to add an L-Glutamine outflux reaction. This will allow you to explore the process and familiarize yourself with adding reactions to the network. It's worth noting that this program contains many more features beyond the scope of this basic tutorial. If you're interested, I recommend reading through the instruction manual to discover additional features that can further customize your networks and enhance your experience with the tool. </li>
										<br>
										<img src="images/Screenshot from 2022-09-28 01-39-52.png" width=750 />
										<br><br>
									</ol>
								
								 </li>
								 
								 <li><strong>Part 6: Finalizing data visualization to the network:</strong> 
									<ol>
										<br>
										<img src="images/Screenshot from 2023-07-31 11-09-06.png" width=750 /> <br><br>
										<li>
										Users can adjust the font size of the labels by selecting the font menu option and sliding the smaller bar at the bottom. Users can adjust the numerical range on both sides which can go upto +/-1000. 
										</li>
										<img src="images/Screenshot from 2023-07-31 11-10-03.png" width=750 /> <br><br>
										<li>
										Users can delete unwanted nodes in the network by first selecting specific nodes and then delete->only selected. 
										</li>
										<img src="images/Screenshot from 2023-07-31 11-10-52.png" width=750 /> <br><br>
										<img src="images/Screenshot from 2023-07-31 11-11-32.png" width=750 /> <br><br>
										<li>
										Users can turn grey links to dark by highlighting them. First, select nodes (both metabolite nodes and reactions nodes) that needed to be highlighted. Then in the menu under Geometric operations->Highlight.
										</li>
										<img src="images/Screenshot from 2023-07-31 11-12-05.png" width=750 /> <br><br>
										<img src="images/Screenshot from 2023-07-31 11-13-35.png" width=750 /> <br><br>
										<li>
										Users can also make circular arcs by using the make arc function. First, select only nodes (both metabolite nodes and reactions nodes) that you want to be connected using arc. Then go to the menu under Geometric operations->Make arc.
										</li>
										<img src="images/Screenshot from 2023-07-31 11-25-04.png" width=750 /> <br><br>
										<li>
										Users can then add the additional transport reactions as shown earlier.
										</li>
										<img src="images/Screenshot from 2023-07-31 11-29-57.png" width=750 /> <br><br>
										<li>
										Users can add a rectangle to represent a cell membrane or wall. This option appears in the insert->rectangle. The newly added rectangle will be add the top of the network which will hide the network. To push it back, go to the Geometric operations->Layer->Push to back. This will push the rectangle behind the network. Additional attributes of the rectangle can be adjusted such as roundness of corners, and border thickness from the menu which had a font option. The range can be adjusted as usual. To change the background color, first select the rectangle. The selected rectangle will look like green with two red circles on the diagonal. These circles allow the user to adjust the size of the rectangle. Once the rectangle is selected, slide the black strip on the color scale. Users can choose colors from various color scales in the color scales menu at the bottom.
										</li>
										<img src="images/Screenshot from 2023-07-31 11-33-04.png" width=750 /> <br><br>
										<li>
										This is the final network visualization of the data looks like.
										</li>
									</ol>
								 </li>
								
							</ul>
							
							<br><br>
							 The tutorial above provides detailed instructions on how to showcase a sample dataset that includes intracellular, extracellular, and flux metabolomics data from two organisms: Staphylococcus aureus (SA) and Pseudomonas aeruginosa (PA) on a network. However, the potential for extracting valuable insights goes beyond this basic demonstration of data visualization on the network. To explore further, follow the steps outlined below.
							 
							 <br><br>
							Sample network file following the above instructions:
							<a href="KEGG_pae_Arginine biosynthesis_sample_network.json" download target="_blank">
							  Download the sample network file
							</a>
							 
							<br><br>
							Suppose, instead of having data for multiple time points, the user has data for a single time point. In such cases, the user can utilize the provided sample data file, where duplicate data is used for different time points. This allows for the exploration of the data even with a single time point dataset. 
							<a href="2022_07_15_MINNO_Supplemental_Sample_Data2.csv" download>
							  Download sample dataset2
							</a>
							. This data file can be utilized with the provided sample network file. 
							
							<br><br>
							<a class="offset" name="gene_knockout"></a>
							<strong3>Gene Knockout:</strong3>
							<br><br>
							
							<ul>
								<li><strong>Repeat parts from 1 to 4:</strong> 
									<br>
									
									<ol>
										<br>
										<li><a href="2022_07_15_MINNO_Supplemental_Gene_knockout_Sample_Data.csv" download>
										  Download the gene knockout sample dataset.
											</a>
										</li>
										<br>
										<li><a href="Central_metabolism_gene_knockout.json" download target="_blank">
										  Download the sample network.
											</a>
										</li>
										<br>
										<li>Follow the same steps for uploading the data. If using the above-provided sample network file, the rectangle will be added on the top of the network which will hide the network. To push it back, go to the Geometric operations->Layer->Push to back. This will push the rectangle behind the network. This is an empirical dataset just to show some of the applications of MINNO besides metabolomics data. Use the None Method and scaling value to 1. Below network visualization shows wild type say E.coli metabolism.</li> 
										<br>
										<img src="images/Screenshot from 2023-07-31 22-55-14.png" width=750 />
										<br><br>
										<li> 
										This one when Electron transport chain reactions have been knockout or under anaerobic conditions, the cell will produce lactate to recover NAD+ from NADH generated from glycolysis. Overall, due to less energy production, the growth rate will slow down as appears from the thickness of the edges.	
										</li>
										<br>
										<img src="images/Screenshot from 2023-07-31 22-55-26.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								
							</ul>
							
							<br><br>
							<a class="offset" name="protein_exp"></a>
							<strong3>Protein Expression:</strong3>
							<br><br>
							
							<ul>
								<li><strong>Repeat steps from 1 to 5:</strong> 
									<br>
									
									<ol>
										<br>
										<li><a href="2022_07_15_MINNO_Supplemental_Protein_expression_Sample_Data.csv" download>
										  Download the protein expression sample dataset.
											</a>
										</li>
										<br>
										<li><a href="Central_metabolism.json" download target="_blank">
										  Download the sample network.
											</a>
										</li>
										<br>
										<li>Follow the same steps for uploading the data. If using the above-provided sample network file, the rectangle will be added on the top of the network which will hide the network. To push it back, go to the Geometric operations->Layer->Push to back. This will push the rectangle behind the network. This is an empirical dataset just to show some of the applications of MINNO besides metabolomics data. Use the None method and scaling value set to 2. High expression of glucose transport proteins leads to higher flux through central carbon metabolism. </li> 
										<br>
										<img src="images/Screenshot from 2023-07-31 23-28-59.png" width=750 />
										<br><br>
										<li> On the contrary, low expression of glucose transport protein will reduce the flux through the central carbon metabolism.</li>
										<br>
										<img src="images/Screenshot from 2023-07-31 23-28-49.png" width=750 />
										<br><br>
									</ol>
								</li>
								
								
							</ul>
							
							<br><br>
							<a class="offset" name="crossfeeding"></a>
							<strong3>Cross-feeding:</strong3>
							<br><br>
							
							<ul>
								<li><strong>Repeat steps from 1 to 5:</strong> 
									<br>
									
									<ol>
										<br>
										<li><a href="2022_07_15_MINNO_Supplemental_sp1_Sample_Data.csv" download>
										  Download sample dataset for species 1.
											</a>
										</li>
										<br>
										<li><a href="Central_metabolism_sp1.json" download target="_blank">
										  Download the sample network for species 1.
											</a>
										</li>
										<br>
										
										<br>
										<li><a href="2022_07_15_MINNO_Supplemental_sp2_Sample_Data.csv" download>
										  Download sample dataset for species 2.
											</a>
										</li>
										<br>
										<li><a href="Central_metabolism_sp2.json" download target="_blank">
										  Download the sample network for species 2.
											</a>
										</li>
										<br>
										
										<br>
										<li><a href="2022_07_15_MINNO_Supplemental_cross_feeding_Sample_Data.csv" download>
										  Download sample dataset for cross-feeding.
											</a>
										</li>
										<br>
										<li><a href="Central_metabolism_cross_feeding.json" download target="_blank">
										  Download the sample network for cross-feeding.
											</a>
										</li>
										<br>
										<li>Follow the same steps for uploading the data. If using the above-provided sample network file, the rectangle will be added on the top of the network which will hide the network. To push it back, go to the Geometric operations->Layer->Push to back. This will push the rectangle behind the network. This is an empirical dataset just to show some of the applications of MINNO besides metabolomics data. Use the None Method and scaling value to 1 for all three cases. Metabolic data for species 1</li>
										<img src="images/Screenshot from 2023-08-01 17-35-53.png" width=750 />
										<br><br>
										<li>Metabolic data for species 2.</li>
										<br>
										<img src="images/Screenshot from 2023-08-01 17-35-21.png" width=750 />
										<br><br>
										<li>Metabolic data for both species 1 and species 2 interacting with each other.</li>
										<br>
										<img src="images/Screenshot from 2023-08-01 17-38-59.png" width=800 />
										<br><br>
									</ol>
								</li>
								
							</ul>
							
							
						</font>
					</p>
				
					
					<p>
					 Troubleshooting 
					 Refresh your browser
					 <em> We recommended saving many versions of your network as you make changes and then refresh/upload the new version to avoid losing network data and to decrease the lag time when adding new features to the network.</em>
					</p>
					
					<p>
						<font face="Arial" size="3">
							<a class="offset" name="upload"></a><strong2>Upload Data:</strong2> 
							<ol>
										<li><strong>Network file:</strong> Import network file. You can upload multiple network files and the tool will combine all the network file together. The tools also has a feature to load kegg base/ortholog map files along with organism specific map together. This will allow user to showcase organism specific rxns on the KEGG base metabolic pathway maps. As all these loaded files are in json format, in order to make distinction between them, they need to have specific prefixs to their names. The steps involved are mention below:
										<br>
										<ul>
											<br>
											<li>You will need to upload two different kinds of files together. The one file with the base network layout must have the prefix as Ortholog_ko_... .json. Here ... fills any specific pathway name such as Glycolysis, TCA etc. While saving files, the tool automatically suggest suitable prefix for the file. 
											</li>
											<br>
											<li> The second file name must be prefixed with KEGG_org_... .json where ‘org’ is the id of the specific organism to specify its metabolic rxns. While saving files, the tool automatically suggest suitable prefix for the file.
											</li> 
											<br>
											<li>All these maps both Ortholog and KEGG can be obtained by the selecting organism from the kegg list and pathway name from the KEGG database menu. Similarly, you can get Ortholog files by searching for ortholog in the KEGG organism menu.									
											</li>
											<br>
											<li>Both these files must have the same pathway as the visualization code matches the rxns ids in the organism-specific .json file with the base .json file. You can load multiple pathway files for both sets i.e., Ortholog_ and KEGG_ as mentioned earlier to get combined network consisting of multiple pathways together. To avoid overlap between them, you can show grid and spatially organize the individual pathways.
											</li>
											<br>
											<li> Adding rxns to specific organisms is simple. Just add specific rxn to the base/ortholog map. 
											</li>
											<br>
										</ul>
										
										</li>
										<br>
										<li><strong>Data file:</strong> Importing data file which will be shown onto the network.
										 </li>
										<br>
										<li><strong>Images:</strong> It allows the import of all kind of images such as png, svg, jpeg, jpg.						
										</li>								
								</ol>
								
								<a class="offset" name="download"></a>
								<strong2>Download data:</strong2>
								<ol>
								
									<li><strong>Network:</strong> 
										<ul>
											<br>
											<li><strong>JSON:</strong> Downloading the network file in json format.
											</li>
											<br>
											<li><strong>KGML:</strong> Downloads network file in KEGG kgml xml-format. This feature allow share of new discoveries to the central KEGG database.
											</li>
											<br>
											<li><strong>PNG:</strong> 
											Download png file of the visualization.
											</li>
											<br>
											<li><strong>SVG:</strong> 
											Download SVG file of the visualization.
											</li>
											<br>
										</ul>
									</li>
										
									<li><strong>Datafile:</strong> 
										<ul>
											<br>
											<li><strong>JSON:</strong> 
											Download data file in json format.
											</li>
											<br>
											<li><strong>CSV:</strong> 
											Download datafile in CSV format.
											</li>
											<br>
										</ul>
									</li>
								</ol>
						</p>
						
						<br><br>	
						<a class="offset" name="schema"></a>
						<strong2>Datafile Schema:</strong2></a> 
						<br><br>
						The data file can be either .csv or .json format to be loaded into the visualization code. Each data file format has information of variables/conditions which are varied and quantitative measurements of various biomolecules which are represented as nodes or edges in the network. Below are the steps to make datafile:
						<br><br>
						
						<ol>
							<li>
								 Assuming three variables which are varied to be ‘X’, ‘Y’, ‘Z’ where 'X', 'Y' and 'Z' can be any arbitrary variables such as organism name, chemical treatment etc. If there are no three parameters for any experiment data, then just define arbitrary variable and give a constant value to it. Here is what schema the datafile must follow.
							
							</li>
							
							<li>
								<strong>JSON:</strong>
								Assuming same three variables which we have used above, the data should be organized as shown below:<br>
								{‘X1’:<br>
								{‘Y1’:[‘name’: ___, ‘gp’:____, ‘Z1’:___, ‘Z2’:____, ‘Z3’:____ .....],<br>
								‘Y2’:[‘name’: ___, ‘gp’:____, ‘Z1’:___, ‘Z2’:____, ‘Z3’:____ .....],<br>
								.<br>
								.<br>
								.<br>
								}<br>
								‘X2’:<br>
								{‘Y1’:[‘name’: ___, ‘gp’:____, ‘Z1’:___, ‘Z2’:____, ‘Z3’:____ .....],<br>
								‘Y2’:[‘name’: ___, ‘gp’:____, ‘Z1’:___, ‘Z2’:____, ‘Z3’:____ .....],<br>
								..<br>
								.<br>
								},<br>
								.<br>
								.<br>
								.<br>
								}<br>
								
								<br><br>
								Above X1, X2, Y1, Y2, Z1, Z2... are different X, Y, and Z variables respectively as there can be an arbitrary number of these variables in a given dataset. In addition to these three variables, we also have ‘name’ to specifically locate node in the network and ‘gp’ for group of dataset that one wants to be loaded onto the node. This allow same node name to represent multiple dataset depending on what it represents in a given network. For instance, if one is trying to visualize glucose uptake by E.coli, then one need to specify extracellular and intracellular signal intensity for glucose and how fast signal intensity is changing provide flux value. This way, same glucose node in the network can show different dataset. Generally biological experiments have control experiment which provides a way to compare and relate how certain perturbation has impacted the experimental results. This control dataset is also need to be included in the above both datafile formats and must be represented as ‘C’ in the datafile as a variable that is varied during the experiment.
							</li>
							
							<li>
								<strong>CSV:</strong> CSV format is easier to create using Microsoft Excel or Libre Office. The CSV file must have the following header<br><br>
								
								Compound, Group, X1_Y1_Z1, X1_Y1_Z2,....<br>
									A ,    b,     1.0,     2.0   .... <br>
									.<br>
									.<br>
								<br><br>
								Here, X1_Y1_Z1 represents three variables as a single string which are tied together using an underscore character. Each line in the CSV file has a compound name, its group id for example whether the compound represents "I" (intracellular) or "E" (extracellular) molecule etc, and quantitative measurement for X, Y and Z variables. Except for first two column, the rest of the header does not need to be in any particular order as long as all of them are included. Generally biological experiments have control experiment which provides a way to compare and relate how certain perturbation has impacted the experimental results. This control dataset is also need to be included in the above both datafile formats and must be represented as "C" in the datafile as a variable that is varied during the experiment.
							</li>
						
						</ol>
						<br><br>
						<p><a class="offset" name="visual"></a>
							<strong2>Visualization</strong2>
							<br><br>
							<strong>Load data file and its visualization onto the network:</strong>
							
							To load datafile and visualize it onto the network, follow the following steps:
							<font face="Arial" size="3">
								<ol>
									<br>
									<li>After loading the network file load the data file and then the user must assign a group to nodes before normalization. Select Assign group menu and select a specific type of node and assign all the nodes to the desired group from the menu. </li>
									<br>
									<li>Then comes applying methods. Each group of datasets needs to be uploaded onto the network can have different normalization method. Often raw data contains big numbers which cannot be loaded onto the network. Thus, normalization and scaling are required. Select the suitable method for specific group and adjust scaling parameter value using the big slider in the middle. Do this for all groups of data whether it is intracellular, extracellular or edges of the network. 
									</li>
									<br>
									<li>It is done. Now change the datagroups from the top 3 menus which update the network as per values for 3 variables you provided in the datafile. 
									</li>
								</ol>
							
							</font>
						</p>
						
						<p>
						<font face="Arial" size="3"> <a class="offset" name="normalization"></a>
							
							<strong>Data Normalization:</strong> The data normalization methods appear in the "Method" drop-down menu after data file is loaded. MINNO allows users to define groups to different nodes of the network. Subsequently, data from the data file will be expressed onto nodes as per their group defined by the user. Each group from the data file can be treated individually depending on which method is suitable for them. Four normalization methods are provided by the tool:
							
							<ol>
								<li><strong>None:</strong> This method allow data values to be loaded and used as they are in the data file onto the network nodes. No normalization method is implemented in this option.</li>
								<br>
								<li><strong>Control normalization:</strong> This method computes the standard deviation of the control dataset for each compound. Then it subtracts the value measured in certain condition from the value in the control case for each metabolite and specific time point and divides the difference with the standard deviation to get normalized value.
								<br><br>
								Let std_dev be the standard deviation of the control dataset. Then the normalized data can be obtained using the below eq.<br><br>
								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Ny(t) = (y(t) – c(t))/std_dev <br><br>
								here Ny(t) is the normalized value of y(t) from Y dataset at time t when it is compared with the control value via c(t) at the same time.
								 </li>
								<br>
								<li><strong>Control Mean normalization:</strong> This method computes the mean and standard deviation of the control dataset for each compound. Then it subtracts the value measured in a certain condition at a specific time point and the mean value of the control dataset and divides the difference with the standard deviation to get a normalized value.
								<br><br>
								
								Let c_mean and std_dev be the mean and standard deviation of the control dataset. Then the normalized data can be obtained using the below eq. <br><br>
								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Ny(t) = (y(t) – c_mean)/std_dev <br><br>
								here Ny(t) is the normalized value of y(t) from Y dataset at time t when it is compared with the mean value of the control dataset C for all time points.
								
								</li>
								<br>
								<li><strong> Log2 fold change:</strong> This method computes log(y(t)/c(t)) value for Y dataset at time t with respect to control dataset c(t) at time point 't' for each compound.</li>
							</ol>
						
						</font>
					</p>
					
					<p>
						<font face="Arial" size="3">
							<strong>Parameter:</strong> By default the tool uses a constant base value and allows user to decide the scaling value to be used for any given group within the datafile. The scaling value essentially allow how much one unit difference in the normalization data would translate into node sizes. Scaling value >1 means that the 1 unit difference in the normalized data would translate into >1 unit difference in the node size. Such value of scaling to be used when the variation in the normalized data values is small and hard to distinguish visually. On the other hand it is possible that the datafile contain values which have large variation such that everything takes extreme values i.e., nodes appears either giant or point size. In such situation, we need to use scaling <1 to diminish the variation is certain group/s and focus on the relevant ones which do have extremely large or small values. The value of the scaling parameter can be adjusted by using the bigger slider above PNG download button. Each time slider value is adjusted, normalization of only selected group changes. Keep doing for all the different groups you have in the datafile. This way user can normalized wide variety of data specifically.
							
							<br><br>
							
							<strong>Assign Group:</strong> This menu contains options for different groups that can be assigned to network nodes. This menu is updated when datafile is uploaded in the web-application. To assign specific group to any node, one first need to select all nodes which belongs to the specific group such as intracellular or extracellular and then select the group from the Assign group drop-down menu. After assigning different groups to nodes, user must select the method to each group and scale the values using scaling option. At first user must select scaling to be 1. Later user can change its value depending on need.
							
							<br><br>
							
							<strong>Hide Menu:</strong> This menu offers options to hide certain aspects of the visualization. It has the following options:
						</font>
					</p>
						<br><br>
						<a class="offset" name="features"></a>
						<strong2>Additional features</strong2>
						<br><br>
						It offers multitudes of features which are listed below:<br><br>
						
						<strong>Simulation:</strong> This key will allow the self-organization of nodes in the network based on force simulation where nodes in the network repel each other while edges attract them together. This is useful when the network does not come with initial coordinates such as metabolic networks from <a href="http://bigg.ucsd.edu/" target="_blank"><span>BiGG-models database</span></a> However, if the network is already pre-layout such as network from KEGG database, then keeping simulation switched off will make network nodes static.
						
						<br><br>
						
						<strong>Freeze:</strong> This feature is added to improve the user experience while using the tool in the case  when the network is significantly big. However, having too big files is not recommended as it can crash the browsers. This web application uses the D3 JavaScript library as underlying program to layout the network visualization. In case of big files, this can take long time to execute and user actions will changes/updates the visualization slowly. By clicking it on, user can stop the underlying computation early hence improving the user experience. But this also comes with a cost as it prevent dragging of network elements jerky or even make them hard to move and some part of network may not even update completely.
						
						<br><br>
						
						<strong>Three data groups:</strong> These three drop-down menus will be containing 3 variables from the data file. These 3 variables are variables that can be varied within an experiment or processed data. For example, time is one commonly used variable in any dataset. The other two variables could be two kinds of treatments or different Omic data and one is interested in combination of these two variables together. Thus, having 3 variables for this tool expand its application to larger variety of experiments. The datafile can be either csv file or json file. <a href="#schema"><span>Click here for more details.</span></a>
						
						<br><br>
						
						<strong>KEGG organisms:</strong> This drop-down menu offers a list of all organisms listed in the KEGG database. To obtain species-specific metabolic pathways, select all the organisms for which you need metabolic pathways. Then select the metabolic pathway from the "KEGG database" menu under "list of pathway" category.<a href="http://rest.kegg.jp/list/organism" target="_blank"><span>Click here for list of all organisms.</span></a>
						
						<br><br>
						
						<strong>Select Rxns:</strong> Once a network file is loaded, the list of reactions will appear here. Users can select all or specific ones by scrolling over the list.
						
						<br><br>
						
						<strong>Remove element:</strong>This menu contains a list of nodes sorted based on their degree in the network. By selecting elements from the menu, these elements will be removed globally from the metabolic network. 
						
						<br><br>
						
						<strong>Repeat element:</strong> This menu contains a list of nodes sorted based on their degree in the network. By selecting elements from the menu, these elements will be duplicated globally from the metabolic network. 
						
						<br><br>
						
						<strong>Find element:</strong> This menu contains a list of nodes in the network sorted alphabetically. Users can use this list to find specific elements in the network.
						
						</font>
					</p>

					<p>
						<font face="Arial" size="3">
							In the menu right to the Assign-group menu, there are few more features as discussed below:
							<ol>
								<br>
								<li><strong>Grid:</strong> This feature shows/hides the grid in the visualization. The purpose of a grid is to offer a way to spatially organize a network by combing different network files.</li>
								<br>
								<li><strong>Tooltip:</strong> This feature shows details associated with individual elements of the network. For KEGG metabolic networks, this also offers more options to explore metabolite, rxn, and gene details from the KEGG database.  
								</li>
								<br>
								<li><strong> Reaction node name:</strong> This feature show and hide the name of reaction nodes of the metabolic network. These reaction nodes also called link nodes connect various metabolite nodes together to represent a chemical reaction.
								</li>
								<br>
								<li><strong> Link node:</strong> Show or hide the link node.
								</li>
							</ol>
						
						</font>

						<font face="Arial" size="3">
							<strong>Attribute Menu:</strong> This provides all the options that the user can vary as desired. They are grouped into text, rectangle, spacing, node, link, image resolution, and freeze degree. Users can vary various attributes by adjusting smaller slider below the menu. On either side of slider, there are inputs where user can provide values to adjust the range of the slider for different attributes as required. There is a limit of +-1000 units value. Anything above or below absolute 1000 units will set default to 1000 value. User can select network elements on which such modifications can be implemented by selecting them by either clicking on them individually or using shift key with mouse drag to scan the area interested.
							
							<ol>
								<br>
								<li><strong>Text:</strong> Provides options for changing font-size, x and y position of text relative to node position.</li>
								<br>
								<li><strong>Rectangle:</strong> Provides options for changing the width, height, border thickness, and corner roundness of the selected rectangles. 
								</li>
								<br>
								<li><strong> Spacing:</strong> Provides options for changing linear spacing, and circular spacing between elements of the network.
								</li>
								<br>
								<li><strong>Node:</strong> Changes the size of the node.
								</li>
								<br>
								<li><strong>Link:</strong> Changes the thickness of the link/edges.
								</li>
								<br>
								<li><strong>Image resolution:</strong> Allow user to vary image resolution for the png image download as desired. While if there is an image imported into the visualization then the option to change the dimension of the image file while maintaining the aspect ratio of the selected image is the same. 
								</li>
								<br>
								<li><strong>Freeze degree:</strong> This option control how much internal computation is for simulation or user interaction such as dragging the user wants to compromise. A magnitude closer to 1 prevents all the dragging of network elements. More precisely, elements can still move but their new position will not update immediately. However, when the user wants to move some other element, previously moved elements will have a new position. If the magnitude value is reduced to 0.5-0.3, the user can get dragging behavior but may not be as smooth as you will have if you choose a magnitude value of 0.01. As you will notice with big files, having a very small value for freeze makes everything very slow as internal computation takes time to update changes users made. Thus this feature improves user experience by avoiding unnecessary computation.
								</li>
								<br>
							</ol>
						
						</font>
					</p>
					
					<p>
						<font face="Arial" size="3">
							<strong>Color Scale Menu:</strong> D3 library provides several sequential, diverging color scales with single or multiple hues. <a href="https://github.com/d3/d3-scale-chromatic" target="_blank"><span>Click here for more detail.</span></a> 
							
							<br><br>
							
							<strong>Invert:</strong> Invert the color scale. 
							
							<br><br>
							
							<strong>Color scale:</strong> This shows the color scale chosen from the Color Scale Menu. It also offers a thin black strip which can be dragged using mouse. User can select network element and set the color by dragging the black strip.
						</font>
					</p>
					
					<p>
						<font face="Arial" size="3"> <a class="offset" name="edit_tools"></a>
						
						<br><br>
						<strong2>Edit tools</strong2>
						<br><br>
						
						In addition to above features, the web-application has another menu which pop-up on mouse right click on the visualization screen. It offers options like file upload to download into various formats and various geometric operation to make organization of network elements easier and faster.
						
						<br><br> 
						
						Below we provide the list of options provided in the pop-up menu:
				
							<ol>																
								<li><strong>Insert:</strong> 						
									<ul>
										<br>
										<li><strong>Text:</strong> 
										Add arbitrary text to the visualization to add the necessary information. The font size and color can be adjusted using the attribute Menu discussed above.
										</li>
										<br>
										<li><strong>Reaction:</strong> 
										Add two nodes connected by a directed edge.
										</li>
										<br>
										<li><strong>Node:</strong> 					
										Add a node element of the network. Note: The user cannot connect nodes together and build the network as the tool will exclude such connections while uploading network files. The node must connect to the link node and vice versa.
										</li>
										<br>
										<li><strong>Link node:</strong> 
										Add a link node of the network. This allows connecting two network elements. Note: The user cannot connect nodes together and build the network as the tool will exclude such connections while uploading network file. The node must connect to link node and vice versa.
										</li>
										<br>
										<li><strong>Rectangle:</strong> 
										Add a rectangle to the visualization. Its attributes such as border thickness, roundness, and color can be adjusted using the attribute menu discussed above.
										</li>
										<br>
										<li><strong>Color scale:</strong> 						
										Add a color scale that appears on the bottom of the taskbar. This scale allows the interpretation of data when loaded onto the network.
										</li>
										<br>
									</ul>
								</li>
								
								<li><strong>Select:</strong> 						
									<ul>
										<br>
										<li><strong>Node:</strong> 
										Select only nodes which were scanned using the shift key.
										</li>
										<br>
										<li><strong>Link node:</strong> 
										Select only link nodes that were scanned using the shift key.
										</li>
										<br>
										<li><strong>Link:</strong>
										Select links connected to selected nodes that were scanned using the shift key.
										 </li>
										<br>
										<li><strong>Both nodes and links:</strong> 
										Select both nodes and links which were scanned using the shift key.						
										</li>
										<br>
									</ul>
								</li>
								
								<li><strong>Delete:</strong> 						
									<ul>
										<br>
										<li><strong>Only selected:</strong> 
										Delete only selected network element/s or other elements created in the visualization.
										</li>
										<br>
										<li><strong>All:</strong> 
										Delete everything on the visualization screen.
										</li>
										<br>
										<li><strong>Color scale:</strong> 						
										Delete color scale which can be added using the insert option.
										</li>
									</ul>
								</li>
								<br>
								<li><strong>Geometric Operations:</strong> 						
									<ul>
										<br>
										<li><strong>Alignment:</strong>
											<ul>
												<br>
												<li><strong>Horizontal:</strong> 
												Horizontal alignment of selected network elements. Users can use this option to align selected nodes, rectangles, or images used in the visualization.
												</li>
												<br>
												<li><strong>Vertical:</strong> 
												Vertical alignment of selected network elements. Users can use this option to align selected nodes, rectangles, or images used in the visualization.
												</li>
												<br>
												<li><strong>Circular:</strong> 
												Circular alignment of selected network elements. This form an equispaced ring. It is important to select network elements that the user wants to appear as a ring.
												</li>
												<br>
											</ul>
										</li>
										<br>
										<li><strong>Layers:</strong> 
											This web application creates various svg elements and groups them in classes. For example, the nodes are grouped under the nodes group, links are grouped under the links group, rectangles are grouped under the rectangles group etc. How and when a certain group is created decides which elements will come on top of others. However, users may prefer to reorder these groups to build a more appealing visualization. To do that, the application offers a way to push different groups at the top or bottom.  
											<br>
											<ul>
												<br>
												<li><strong>Bring to front:</strong> 
												Bring an entire group of elements to the top.
												</li>
												<br>
												<li><strong>Push it back:</strong> 
												Pushes all the elements of the group at the bottom.
												</li>
											</ul>
											<br>
											These options change the order of groups but not the order of elements within individual groups. This way user can push the rectangle svg group at the bottom so that the network can be visualized on top of it. However, when multiple rectangles are created, users need to manage how different rectangles will order themselves based on the order of their creations. 
										</li>
										<br>
										<li><strong>Duplicate:</strong> 
										This option allows the user to make duplicates of selected network elements. Using this, certain parts of the network can be duplicated and its nodes and links can be colored or changed their size to highlight different groups of data. This feature is basically to create a simplified network representation to compare different datasets.
										</li>
										<br>
										<li><strong>Rotation:</strong> 				
										Rotate selected network elements about their centroid.		
										</li>
										<br>
										<li><strong>Make arcs:</strong> 						
										Make curved edges. For example, it makes nice curved edges for circular loops in the network.
										</li>
										<br>
										<li><strong>Remove arcs:</strong>
										Revert back curved edges to straight. 						
										</li>
										<br>
										<li><strong>Highlight:</strong>
										Change the opacity to 1.0 for selected network elements. This feature also allows the user to select which reactions will be selected to be included in the KGML file of the network. Only nodes and edges whose opacity is 1.0 will be included in the KGML file.						
										</li>
										<br>
										<li><strong>Un-Highlight:</strong> 					
										Change the opacity to 0.5 for selected network elements. This feature also allows the user to select which reactions will be selected to be included in the KGML file of the network. Nodes and edges whose opacity is 0.5 will be excluded in the KGML file.												
										</li>
									</ul>
								</li>
								<br>
								<li><strong>Undo:</strong>
								It only undo nodes' movements or dragging events by the user. If you delete or insert any network element, Undo cannot revert to the previous network. 
								</li>
								<br>
								<li><strong>Redo:</strong>
								It only redo nodes movements or dragging events by the user. If you delete or insert any network element, Undo cannot revert to the previous network.
								</li>
								<br>
								<li><strong>Help:</strong>
								Opens up User Manual.
								</li>
							</ol>
						</font>
					</p>
					
					
					<p>
						<a class="offset" name="copyright"></a>
						
						<br><br>
						<strong2>Copyright</strong2>
						
						<font face="Arial" size="3">	
							<br><br><br>
							For any further questions, please contact us at this email address: <strong>ayush.mandwal@ucalgary.ca</strong> 
							<br><br>
							It is made available under a CC-BY-NC-ND 4.0 International license. <br><br>
							
							<strong>Copyright 2021 AYUSH MANDWAL</strong><br><br>

							<img src="by-nc-nd.png" alt="Italian Trulli" style="width:200px;height:70px;">
							
						</font>
					</p>
				
				</p>

    </div>
    <!--end content div-->
  </div>

</div>
<!--end nav-body div-->

</body>

</html>
