<html>

<style>
#nav-body {
	font-family: Arial, Helvetica, sans-serif;
	background-color: white;
	display: inline-block;
	width: 100%;
	vertical-align: top;
	overflow: hidden;
}
#menu {
	margin: 0;
	padding: 0;
	width: 12em;
	background-color: white;
	list-style-image: none;
	list-style-type: none;
	overflow: auto;
	position: relative;
	display: inline-block;
	clear: both;
	float: left;
	width:300px;
}
#menu li {
	font: 90%"Arial", "Bitstream Vera Sans", "Trebuchet Unicode MS", "Lucida Grande", Verdana, Helvetica, sans-serif;
	margin: 0;
	padding: 0;
	list-style: none;
}
#menu a {
	background: white;
	border-bottom: 0px solid #336699;
	color: #000;
	display: block;
	margin: 0;
	padding: 8px 12px;
	text-decoration: none;
	font-weight: normal;
}
#menu a:hover {
	background: #2580a2 url("images/hover.gif") left center no-repeat;
	color: #fff;
	padding-bottom: 8px;
}
#content-wrapper {
	width: auto;
	overflow: hidden;
}

.content {
	font-size: medium;comment in css
	color: black;
	z-index:-1;
	background-color: #FFF;
	display: inline-block;
	width: 100%;
	height: 88%;
	vertical-align: 10%;
	overflow: auto;
	overflow-y: scroll;
	/*
	height:490px;
	*/
}

.content::before1{
		content:"";
		background-size: cover;
		position: absolute;
		top: 0px;
		right: 0px;
		bottom: 0px;
		left: 0px;
		background-image: url("MINNO Logo LRG_centered.svg");
		background-repeat: no-repeat;	
		background-attachment: fixed;
		background-position: center;
		background-size: 800px;
		opacity:0.10;
		z-index: 0;
		
		/*
		background-size: 150px;
		width: 300px;
		height: 300px;
		background-position: 25% 75%;
		background-size: cover;
		*/			
	}
	
	.header img {
		float: left;
		width: 200px;
		height: 100px;
		padding-left: 2%;
		background: #fff;
	}

	.header h1 {
		position: relative;
		top: 18px;
		left: 10px;
	}
	
	.offset:before { 
		display: block; 
		content: " "; 
		height: 00px;      /* Give height of your fixed element */
		margin-top: -0px; /* Give negative margin of your fixed element */    
		visibility: hidden; 
	}
	
	strong2{
	  font-weight:bold;
	  font-size:28px;
	  line-height:22px;
	}

</style>

<body>

<div id="nav-body">
	<div class="header">
		<a href="https://lewisresearchgroup.github.io/MINNO/">
		<img src="MINNO Logo LRG_centered.svg" alt="logo" />
		<h1 align="center">
		<font face="Arial" color="black" size="6"> 
			Metabolic Interactive Nodular Network for Omics <br> User Manual 
		</font>
		</h1>
		</a>
	</div>
	

  <div id="menu">

    <ul>
      <li class='active'><a href='#intro'>Introduction</a>
      </li>
      <li><a href='#databases'>Databases</a>
      </li>
	
      <li><a href='#tutorial'>Tutorial</a>
		<ul>
			<li><a href='#basic'>Basic</a>
			</li>
			<li><a href='#boundary_fluxes'>Boundary fluxes </a>
			</li>
			<li><a href='#additional'>Additional</a>
			</li>
			<!--
			
			<li><a href='#protein_exp'>Protein Expression </a>
			</li>
			<li><a href='#crossfeeding'>Cross-feeding</a>
			</li>
			-->
		</ul>
      </li>
	
      <li><a href='#upload'>Upload data</a>
      </li>
      <li><a href='#download'>Download data</a>
      </li>
      <li><a href='#schema'>Datafile Schema</a>
      </li>
      <li><a href='#visual'>Visualization</a>
      </li>
      <li><a href='#features'>Features</a>
      </li>
      <li><a href='#edit_tools'>Edit tools</a>
      </li>
      <li><a href='#copyright'>Copyright</a>
      </li>
    </ul>
  </div>
  <!--end menu div-->

  <div id="content-wrapper">
    <div class="content" left="50px">
	<p><a class="offset" name="intro"></a>
					<strong2>Introduction:</strong2>
					<br><br>
					&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
					<font face="Arial" size="3">
						
						<strong>Metabolic Interactive Nodular Network for Omics (MINNO)</strong> is a <a href="https://d3js.org/" target="_blank"><span>D3 javascript library</span></a> based web application to visualize biological data onto metabolic networks. Besides visualization, this tools offer various ways to share data and results to facilitate collaboration across multiple research groups globally. Being a web-application, it does not require any installation and it can be run on both Chrome and Mozilla browsers.
						
						<br><br>
						
						The tool accepts any kind of omics data, including metabolomics, proteomics, and fluxomics data as long as they can be visualized by changing sizes of nodes and edges of the metabolic network. The modular design promotes accelerated investigation of omics data on genome-scale metabolic networks for multiple species.
						
						<br><br>
						
						<a class="offset" name="databases"></a> 
						<strong2>Databases:</strong2>
						<br><br>
						MINNO utilizes two biological databases: Kyoto Encyclopedia of Genes and Genomes (KEGG) and National Center for Biotechnology Information- Reference Sequence (NCBI-RefSeq) database.
						
						<ol>
							<li><a href="https://www.kegg.jp/" target="_blank"><span><strong>KEGG-Database:</strong></span></a> Kyoto Encyclopedia of Genes and Genomes (KEGG) is a knowledge base for systematic analysis of gene functions, linking genomic information with higher order functional information stored in the PATHWAY database, which contains graphical representations of cellular processes, such as metabolism, membrane transport, signal transduction and cell cycle. MINNO comes with 66 base metabolic pathways from the KEGG database, which cover all primary metabolic pathways and can be combined together to build any possible large-scale metabolic network as needed.
							</li>
							
							<br>
							
							<li> <a href="https://www.ncbi.nlm.nih.gov/refseq/" target="_blank"><span><strong>NCBI-RefSeq database:</strong></span></a> The NCBI-Reference Sequence (RefSeq) collection provides a comprehensive, integrated, non-redundant, well-annotated set of sequences, including genomic DNA, transcripts, and proteins. It is an open access collection of publicly available nucleotide sequences and their protein products. MINNO enables users to built organism specific metabolic map by constructing large-scale metabolic network using annotated gene sequences from NCBI-RefSeq database.  </li>
							
						</ol>
						
				</p>

	    			<p>
						<font face="Arial" size="3"> <a class="offset" name="tutorial"></a>
						
							<strong2>Tutorial:</strong2>
							<br><br>
							The below instructions provide guidelines for various features that MINNO provides to the users.  <br><br>
							
							<a class="offset" name="basic"></a>
							<strong3>Basic:</strong3>
							<br><br>
							<ul>
								<img src="images/image1.png" width=750 /> <br><br>
								<img src="images/image2.png" width=750 /> <br><br>
								<img src="images/image3.png" width=750 /> <br><br>
								<img src="images/image4.png" width=750 /> <br><br>
								<img src="images/image5.png" width=750 /> <br><br>
								<img src="images/image6.png" width=750 /> <br><br>
								<img src="images/image7.png" width=750 /> <br><br>
								<img src="images/image8.png" width=750 /> <br><br>
								<img src="images/image9.png" width=750 /> <br><br>
								<img src="images/image10.png" width=750 /> <br><br>
								<img src="images/image11.png" width=750 /> <br><br>
								<img src="images/image12.png" width=750 /> <br><br>
								<img src="images/image13.png" width=750 /> <br><br>
							</ul>

							<br><br>
							<a class="offset" name="boundary_fluxes"></a>
							<strong3>Boundary fluxes:</strong3>
							<br><br>

							<ul>
								<ol>
									<li> Boundary fluxes or exchange fluxes across the cell membrane of the cell. The flux profile provides crucial cellular dynamics manifesting the behavior under certain conditions or perturbations. <br>
									<img src="images/Screenshot from 2024-01-05 18-35-08.png" width=750 /> <br><br>
									</li>
									<li>
									Users can use MINNO to show exchange flux data. Here is a sample network file <a href="Arginine_biosynthesis_pathway2.json" download> Download sample network file</a> and <a href="2022_07_15_MINNO_Supplemental_Sample_Data3.csv" download> Download sample data file</a>.
									</li>
									<img src="images/Screenshot from 2024-01-05 18-35-16.png" width=750 /> <br><br>
									<li>
									Users can display the exchange fluxes by adding reactions and renaming link nodes, similar to those in the file. For simplicity, one can choose to give the metabolite name a suffix of 'outflux' and 'influx' to distinguish between influx and outflux. These values can then be normalized from the provided choices of normalization options. Note that, by default, the flux data will be shown by changing the thickness of the link with the name given in the data file. Within the data file, the 'Group' of such entries will be 'link'. Users can have more groups, such as 'extra' (extracellular) or 'intra' (intracellular), for various experimental measurements in a given dataset. Also, note that, by default, metabolites have the type 'M'. If the data file contains groups other than 'M' or 'link', users need to assign the group to specific nodes of interest. To do this, users must select nodes first and then click on a specific category in the 'Assign Group menu'.  
									</li>
								</ol>
							</ul>

							<br><br>
							<a class="offset" name="additional"></a>
							<strong3>Additional:</strong3>
							<br><br>
							
							<ul>
								<ol>
									<br>
									<img src="images/Screenshot from 2023-12-14 10-23-05.png" width=750 /> <br><br>
									<li>
									Users can adjust the font size of the labels by selecting the nodes and then selecting the font menu option and sliding the smaller bar at the bottom. The numerical range on both sides can go up to +/-1000. 
									</li>
									<img src="images/Screenshot from 2023-12-14 10-23-25.png" width=750 /> <br><br>
									<li>
									Users can delete unwanted nodes in the network by first selecting specific nodes and then right-click on the screen for the menu and delete->only selected. 
									</li>
									<img src="images/Screenshot from 2023-12-14 10-24-39.png" width=750 /> <br><br>
									<li>
									Users can also make circular arcs by using the make arc function. First, select only nodes (both metabolite nodes and reactions nodes) that you want to be connected using arc. Then go to the menu under Geometric operations->Make arc.
									</li>
									<img src="images/Screenshot from 2023-12-14 10-25-34.png" width=750 /> <br><br>
									<li>
									Users can add a rectangle to represent a cell membrane or wall. This option appears in the insert->rectangle. The newly added rectangle will be added to the top of the network which will hide the network. To push it back, go to the Geometric operations->Layer->Push to back. This will push the rectangle behind the network. Additional attributes of the rectangle can be adjusted such as the roundness of corners, border thickness, etc from the attribute menu above the color scale menu. To change the background color, first select the rectangle. The selected rectangle will look green with two red circles appearing diagonally. These circles allow the user to adjust the size of the rectangle. Once the rectangle is selected, slide the black strip on the color scale. Users can choose colors from various color scales in the color scales menu at the bottom.
									</li>
									<img src="images/Screenshot from 2023-12-14 10-27-17.png" width=750 /> <br><br>
									<li>
									This is the final network visualization of the data looks like.
									</li>
								</ol>
							</ul>
						</font>
				</p>
				
				
				<p>
					<font face="Arial" size="3">
						<a class="offset" name="upload"></a><strong2>Upload Data:</strong2> 
						<ol>
							
								<li><strong>Network:</strong> 
									<ul>
										<br>
										<li><strong>JSON:</strong> 
										</li>
										<br>
									</ul>
								</li>
									
								<li><strong>Datafile:</strong> 
									<ul>
										<br>
										<li><strong>CSV:</strong> 
										</li>
										<br>
									</ul>
								</li>
							</ol>
							
							<a class="offset" name="download"></a>
							<strong2>Download data:</strong2>
							<ol>
							
								<li><strong>Network:</strong> 
									<ul>
										<br>
										<li><strong>JSON:</strong> 
										</li>
										<br>
										<li><strong>PNG:</strong> 
										
										</li>
										<br>
										<li><strong>SVG:</strong> 
										
										</li>
										<br>
									</ul>
								</li>
									
								<li><strong>Datafile:</strong> 
									<ul>
										<br>
										<li><strong>CSV:</strong> 
										</li>
										<br>
									</ul>
								</li>
							</ol>
					</p>
					
					<br><br>	
					<a class="offset" name="schema"></a>
					<strong2>Datafile Schema:</strong2></a> 
					<br><br>
					The datafile needs to be in .csv. Each datafile format have information of variables/conditions which are varied and quantitative measurements of various biomolecules which are represented as nodes or edges in the network. Below are the steps to make datafile:
					<br><br>
					
					<ol>
						<li>
							 Assuming three variables which are varied to be ‘X’, ‘Y’, ‘Z’ where 'X', 'Y' and 'Z' can be any arbitrary variables such as organism name, chemical treatment etc. If there are no three parameters for any experiment data, then just define arbitrary variable and give a constant value to it. Here is what schema the datafile must follow.
						
						</li>
						<br>
						<li>
							<strong>CSV:</strong> CSV format is easier to create using Microsoft Excel or Libre Office. The csv file must have following header<br><br>
							
							Compound, Group, X1_Y1_Z1, X1_Y1_Z2,....<br>
								A ,    b,     1.0,     2.0   .... <br>
								.<br>
								.<br>
							<br><br>
							Here, X1_Y1_Z1 represents three variables as a single string which are tied together using underscore character. Each line in the csv file has a compound name, its group id for example whether the compound represents "I" (intracellular) or "E" (extracellular) molecule etc, and quantitative measurement for X, Y and Z variables. Except for first two column, the rest of the header does not need to be in any particular order as long as all of them are included. Generally biological experiments have control experiment which provides a way to compare and relate how certain perturbation has impacted the experimental results. This control dataset is also need to be included in the above both datafile formats and must be represented as "C" in the datafile as a variable that is varied during the experiment.
						</li>
					
					</ol>
					<br><br>
					<p><a class="offset" name="visual"></a>
						<strong2>Visualization</strong2>
						<br><br>
						
						Simple loading of a network and data file will allow users to visualize data on the network. By default, MINNO uses Control normalization, as mentioned below, for normalization. Users can change it through the Method menu, which offers multiple data normalization methods. Users can also assign a specific group to a certain set of nodes by first selecting nodes and then using the Assign Group menu to assign a specific group to them. 
						
					</p>
					
					<p>
					<font face="Arial" size="3"> <a class="offset" name="normalization"></a>
						
						<strong>Data Normalization:</strong> The data normalization methods appears in the "Method" drop down menu after datafile is loaded. MINNO allow user to define groups to different nodes of the network. Subsequently, data from the datafile will be expressed onto nodes as per their group defined by the user. Each group from the datafile can be treated individually depending on which method is suitable for them. Four normalization methods are provided by the tool:
						
						<ol>
							<li><strong>None:</strong> This method allow data values to be loaded and used as they are in the datafile onto the network nodes. No normalization method is implemented in this option.</li>
							<br>
							<li><strong>Control normalization:</strong> This method computes the standard deviation of the control dataset for each compound. Then it subtract the value measured in certain condition with the value in control case for each metabolite and specific time point and divide the difference with the standard deviation to get normalized value.
							<br><br>
							Let std_dev be the standard deviation of the control dataset. Then the normalized data can be obtained using below eq.<br><br>
							&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Ny(t) = (y(t) – c(t))/std_dev <br><br>
							here Ny(t) is the normalized value of y(t) from Y dataset at time t when it is compared with the control value via c(t) at the same time.
							 </li>
							<br>
							<li><strong>Control Mean normalization:</strong> This method computes the mean and standard deviation of the control dataset for each compound. Then it subtract the value measured in certain condition at specific time point and the mean value of the control dataset and divide the difference with the standard deviation to get normalized value.
							<br><br>
							
							Let c_mean and std_dev be the mean and standard deviation of the control dataset. Then the normalized data can be obtained using below eq. <br><br>
							&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Ny(t) = (y(t) – c_mean)/std_dev <br><br>
							here Ny(t) is the normalized value of y(t) from Y dataset at time t when it is compared with the mean value of the control dataset C for all time point.
							
							</li>
							<br>
							<li><strong> Log2 fold change:</strong> This method computes log(y(t)/c(t)) value for Y dataset at time t with respect to control dataset c(t) at time point 't' for each compound.</li>
						</ol>
					
					</font>
				</p>
				
				<p>
					<font face="Arial" size="3">
						<strong>Parameter:</strong> The scaling value essentially allow how much one unit difference in the normalization data would translate into node sizes. Scaling value >1 means that the 1 unit difference in the normalized data would translate into >1 unit difference in the node size. 
						
						<br><br>
						
						<strong>Assign Group:</strong> This menu contains options for different group that can be assigned to network nodes. To assign specific group to any node, one first need to select all nodes which belongs to the specific group such as intracellular or extracellular and then select the group from the Assign group drop-down menu. After assigning different groups to nodes, user can select the normalization method to each group and scale the values using scaling option.
						
						<br><br>
						
						<strong>Hide Menu:</strong> This menu offers options to hide certain aspect of the visualization. 
					</font>
				</p>
					<br><br>
					<a class="offset" name="features"></a>
					<strong2>Additional features</strong2>
					<br><br>
					It offers multitudes of features which are listed below:<br><br>
					
					<strong>Three data groups:</strong> These three drop-down menu will be containing 3 variables from the datafile. The datafile can be either csv file or json file. <a href="#schema"><span>Click here for more details.</span></a>
					
					<br><br>
					
					<strong>Select Rxns:</strong> Once a network file is loaded, the list of reactions will appears here. User can select all or specific ones by scrolling over the list.
					
					<br><br>
					
					<strong>Remove element:</strong>This menu contains list of nodes sorted based on their degree in the network. By selecting elements from the menu, these elements will be removed globally from the metabolic network. 
					
					<br><br>
					
					<strong>Find element:</strong> This menu contains list of nodes in the network sorted alphabatically. User can use this list to find specific elements in the network.
					
					</font>
				</p>

				<p>
					<font face="Arial" size="3">
						In the menu right to the Assign-group menu, there are few more features as discussed below:
						<ol>
							<br>
							<li><strong>Grid:</strong> This feature shows/hides the grid in the visualization. The purpose of grid is to offer way to spatially organize network via combing different network files.</li>
							<br>
							<li><strong>Tooltip:</strong> This feature shows details associated with individual elements of the network. For KEGG metabolic networks, this also offers more options to explore metabolite, rxn and gene details from the KEGG database.  
							</li>
							<br>
							<li><strong> Reaction node name:</strong> This feature show and hide the name of reaction nodes of the metabolic network. These reactions nodes also called as link nodes connecting various metabolite nodes together to represent a chemical reaction.
							</li>
							<br>
							<li><strong> Link node:</strong> Show or hide the link node.
							</li>
						</ol>
					
					</font>

					<font face="Arial" size="3">
						<strong>Attribute Menu:</strong> This provide all the options that user can vary as desired. They are grouped into text, rectangle, spacing, node,link, image resolution. User can vary various attributes by adjusting smaller slider below the menu. On either side of slider, there are inputs where user can provide values to adjust the range of the slider for different attributes as required. There is a limit of +-1000 units value. Anything above or below absolute 1000 units will set default to 1000 value. 
						
						<ol>
							<br>
							<li><strong>Text:</strong> Provides options for changing font-size, x and y position of text relative to node position.</li>
							<br>
							<li><strong>Rectangle:</strong> Provides options for changing width, height, border thickness, corner roundness of the selected rectangles. 
							</li>
							<br>
							<li><strong> Spacing:</strong> Provides options for changing linear spacing, circular spacing between elements of network.
							</li>
							<br>
							<li><strong>Node:</strong> Changes the size of the node.
							</li>
							<br>
							<li><strong>Link:</strong> Changes the thickness of the link/edges.
							</li>
							<br>
							<li><strong>Image resolution:</strong> Allow user to vary image resolution for the png image download as desired. While if there is a image imported into the visualization then option to change the dimension of image file while maintaining the aspect ratio of the selected image the same. 
							</li>
							<br>
						</ol>
					
					</font>
				</p>
				
				<p>
					<font face="Arial" size="3">
						<strong>Color Scale Menu:</strong> D3 library provides several sequential, diverging color scales with single or multiple hues. <a href="https://github.com/d3/d3-scale-chromatic" target="_blank"><span>Click here for more detail.</span></a> 
						
						<br><br>
						
						<strong>Invert:</strong> Invert the color scale. 
						
						<br><br>
						
						<strong>Color scale:</strong> This shows the color scale chosen from the Color Scale Menu. It also offers a thin black strip which can be dragged using mouse. User can select network element and set the color by dragging the black strip.
					</font>
				</p>
				
				<p>
					<font face="Arial" size="3"> <a class="offset" name="edit_tools"></a>
					
					<br><br>
					<strong2>Edit tools</strong2>
					<br><br>
					
					In addition to above features, the web-application has another menu which pop-up on mouse right click on the visualization screen. It offers options like file download into various formats and various geometric operation to make organization of network elements easier and faster.
					
					<br><br> 
					
					Below we provide the list of options provided in the pop-up menu:
			
						<ol>																
							<li><strong>Insert:</strong> 						
								<ul>
									<br>
									<li><strong>Text:</strong> 
									Add arbitrary text to the visualization to add necessary information. The font size and color can be adjusted using attribute Menu discussed above.
									</li>
									<br>
									<li><strong>Reaction:</strong> 
									Add two nodes connected by a directed edges.
									</li>
									<br>
									<li><strong>Node:</strong> 					
									Add a node element of the network. Note: User cannot connect nodes together and build the network as the tool will exclude such connections while uploading network file. Node must connect to link node and vice versa.
									</li>
									<br>
									<li><strong>Link node:</strong> 
									Add a link node of the network. This allow connecting two network elements. Note: User cannot connect nodes together and build the network as the tool will exclude such connections while uploading network file. Node must connect to link node and vice versa.
									</li>
									<br>
									<li><strong>Rectangle:</strong> 
									Add a rectangle to the visualization. Its attributes such as border thickness, roundness and color can be adjusted using attribute menu discussed above.
									</li>
									<br>
									<li><strong>Color scale:</strong> 						
									Add color scale which appears on the bottom of the taskbar. This scale allow interpretation of data when loaded onto the network.
									</li>
									<br>
								</ul>
							</li>
							
							<li><strong>Select:</strong> 						
								<ul>
									<br>
									<li><strong>Node:</strong> 
									Select only nodes which were scanned using shift key.
									</li>
									<br>
									<li><strong>Link node:</strong> 
									Select only link nodes which were scanned using shift key.
									</li>
									<br>
									<li><strong>Link:</strong>
									Select links connected to selected nodes which were scanned using shift key.
									 </li>
									<br>
									<li><strong>Both nodes and links:</strong> 
									Select both nodes and links which were scanned using shift key.						
									</li>
									<br>
								</ul>
							</li>
							
							<li><strong>Delete:</strong> 						
								<ul>
									<br>
									<li><strong>Only selected:</strong> 
									Delete only selected network element/s or other elements created in the visualization.
									</li>
									<br>
									<li><strong>All:</strong> 
									Delete everything on the visualization screen.
									</li>
									<br>
									<li><strong>Color scale:</strong> 						
									Delete color scale which can be added using insert option.
									</li>
								</ul>
							</li>
							<br>
							<li><strong>Geometric Operations:</strong> 						
								<ul>
									<br>
									<li><strong>Alignment:</strong>
										<ul>
											<br>
											<li><strong>Horizontal:</strong> 
											Horizontal alignment of selected network elements. User can use this option to align selected nodes, rectangles or images used in the visualization.
											</li>
											<br>
											<li><strong>Vertical:</strong> 
											Vertical alignment of selected network elements. User can use this option to align selected nodes, rectangles or images used in the visualization.
											</li>
											<br>
											<li><strong>Circular:</strong> 
											Circular alignment of selected network elements. This form a equi-spaced ring. It is important to select network elements that user wants to appears as a ring.
											</li>
											<br>
										</ul>
									</li>
									<br>
									<li><strong>Layers:</strong> 
										MINNO creates various svg elements and group them in classes. However, user may prefer to reorder these groups to built more appealing visualization. To do that, MINNO offers way to push different group at the top or bottom.  
										<br>
										<ul>
											<br>
											<li><strong>Bring to front:</strong> 
											Bring entire group of elements at the top.
											</li>
											<br>
											<li><strong>Push it back:</strong> 
											Pushes all the elements of the group at the bottom.
											</li>
										</ul>
										<br>
										These options changes the order of groups but not the order of elements within individual group. This way user can push rectangle svg group at the bottom so that the network can be visualized on the top of it. However, when multiple rectangles are created, user need to manage how different rectangles will order themselves based on order of their creations. 
									</li>
									<br>
									<li><strong>Duplicate:</strong> 
									This option allow user to make duplicates of selected network elements. Using this, certain part of the network can be duplicated and its nodes and links can be colored or change their size to highlight different group of data. This feature is basically to create a simplified network representation to compare different datasets.
									</li>
									<br>
									<li><strong>Rotation:</strong> 				
									Rotate selected network elements about their centeriod.		
									</li>
									<br>
									<li><strong>Make arcs:</strong> 						
									Make curved edges. For example, it makes nice curved edges for circular loops in the network.
									</li>
									<br>
									<li><strong>Remove arcs:</strong>
									Revert back curved edges to straight. 						
									</li>
									<br>
									<li><strong>Highlight:</strong>
									Change opacity to 1.0 to selected network elements. This feature also allow user to select which reactions will be selected to be included for the KGML file of the network. Only nodes and edges whose opacity is 1.0 will be included in the KGML file.						
									</li>
									<br>
									<li><strong>Un-Highlight:</strong> 					
									Change opacity to 0.5 to selected network elements. This feature also allow user to select which reactions will be selected to be included for the KGML file of the network. Nodes and edges whose opacity is 0.5 will be excluded in the KGML file.												
									</li>
								</ul>
							</li>
							<br>
							<li><strong>Undo:</strong>
							It only undo nodes movements or dragging event by the user. If you delete or insert any network element, Undo cannot revert to the previous network. 
							</li>
							<br>
							<li><strong>Redo:</strong>
							It only redo nodes movements or dragging event by the user. If you delete or insert any network element, Undo cannot revert to the previous network.
							</li>
							<br>
							<li><strong>Help:</strong>
							Opens up User Manual.
							</li>
						</ol>
					</font>
				</p>
				<p>
					<a class="offset" name="copyright"></a>
					
					<br><br>
					<strong2>Copyright</strong2>
					
					<font face="Arial" size="3">	
						<br><br><br>
						For any further questions, please contact us at this email address: <strong>ayush.mandwal@ucalgary.ca</strong> 
						<br><br>
						It is made available under a CC-BY-NC-ND 4.0 International license. <br><br>
						
						<strong>Copyright 2021 AYUSH MANDWAL</strong><br><br>

						<img src="by-nc-nd.png" alt="Italian Trulli" style="width:200px;height:70px;">
						
					</font>
				</p>
			
			</p>

    </div>
    <!--end content div-->
  </div>

</div>
<!--end nav-body div-->

</body>

</html>
